clang/test/Preprocessor/predefined-macros.c

```diff
@@ -215,12 +215,11 @@
 // CHECK-HIP-DEV: #define __HIP_DEVICE_COMPILE__ 1
 // CHECK-HIP-DEV: #define __HIP__ 1
 
-// RUN: %clang_cc1 %s -E -dM -triple spir64_fpga-unknown-unknown-sycldevice \
-// RUN: -o - \
+// RUN: %clang_cc1 %s -E -dM -fsycl-is-device \
```

Comment:
Do we  have a 'negative' test for this as well?  A situation where it isn't a fpga and it still shows up?

Reply:
>>Do we have a 'negative' test for this as well?

I have added negative test for default. 

Do you mena to check only-fsycl-is-device?

Reply:
Macro will not show up in the situation where it is not fpga and -fsycl-is-device is used. I can add test for it

Reply:
I mean with the triple that is NOT FPGA.  so -triple spir64-sycldevice and a CHECK-NOT for this define.

Reply:
> Macro will not show up in the situation where it is not fpga and -fsycl-is-device is used. I can add test for it

Great! Thats what I'm hoping for.

Reply:
add a negative test for this.